src/vm/vm_core.rs

```diff
@@ -248,7 +248,13 @@ impl VirtualMachine {
     fn is_zero(addr: &MaybeRelocatable) -> Result<bool, VirtualMachineError> {
         match addr {
             MaybeRelocatable::Int(num) => Ok(num.is_zero()),
-            MaybeRelocatable::RelocatableValue(_rel_value) => Err(VirtualMachineError::PureValue),
+            MaybeRelocatable::RelocatableValue(rel_value) => {
+                if rel_value.offset > 0 {
+                    return Ok(false);
+                } else {
+                    Err(VirtualMachineError::PureValue)
+                }
+            }
```

Comment:
Nit (feel free to ignore):
```suggestion
            MaybeRelocatable::RelocatableValue(rel_value) if rel_value.offset > 0 => Ok(false),
            _ => Err(VirtualMachineError::PureValue),
            }
```

Reply:
In any case, at a glance it seems like the expected logic would be:
```suggestion
            MaybeRelocatable::RelocatableValue(rel_value) => Ok(rel_value.offset.is_zero()),
```